android/src/com/mapswithme/maps/maplayer/MapButtonsController.java

```diff
@@ -0,0 +1,234 @@
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import com.mapswithme.maps.R;
+import com.mapswithme.maps.routing.RoutingController;
+import com.mapswithme.maps.routing.SearchWheel;
+import com.mapswithme.maps.widget.menu.MyPositionButton;
+import com.mapswithme.maps.widget.placepage.PlacePageController;
+import com.mapswithme.util.Config;
+import com.mapswithme.util.Graphics;
+import com.mapswithme.util.UiUtils;
+
+public class MapButtonsController
+{
+  @NonNull
+  private final View mButtonsFrame;
+  @NonNull
+  private final View mZoomFrame;
+  @NonNull
+  private final ImageButton mLayersButton;
+  @NonNull
+  private final View myPosition;
+  @NonNull
+  private final View mSearchButtonFrame;
+  @Nullable
+  private final MyPositionButton mNavMyPosition;
+  @NonNull
+  private final MapLayersController mToggleMapLayerController;
+  @NonNull
+  private final SearchWheel mSearchWheel;
+  private final PlacePageController mPlacePageController;
+  private final float mBottomMargin;
+  private final int mButtonWidth;
+  private final float mInitialButtonMargin;
+  private float mTopLimit;
+  private float mContentHeight;
+  private float mContentWidth;
+
+  public MapButtonsController(@NonNull View frame, AppCompatActivity activity, MapButtonClickListener mapButtonClickListener, PlacePageController placePageController)
+  {
+    mButtonsFrame = frame.findViewById(R.id.navigation_buttons_inner);
+    mZoomFrame = frame.findViewById(R.id.zoom_buttons_container);
+    mPlacePageController = placePageController;
+    View zoomInButton = frame.findViewById(R.id.nav_zoom_in);
+    zoomInButton.setOnClickListener((v) -> mapButtonClickListener.onClick(MapButtons.zoomIn));
+    View zoomOutButton = frame.findViewById(R.id.nav_zoom_out);
+    zoomOutButton.setOnClickListener((v) -> mapButtonClickListener.onClick(MapButtons.zoomOut));
+    myPosition = frame.findViewById(R.id.my_position);
+    mNavMyPosition = new MyPositionButton(myPosition, (v) -> mapButtonClickListener.onClick(MapButtons.myPosition));
+
+    mLayersButton = frame.findViewById(R.id.layers_button);
+    mToggleMapLayerController = new MapLayersController(mLayersButton,
+                                                        () -> mapButtonClickListener.onClick(MapButtons.toggleMapLayer), activity);
+
+    mSearchButtonFrame = activity.findViewById(R.id.search_button_frame);
+    mSearchWheel = new SearchWheel(frame, (v) -> mapButtonClickListener.onClick(MapButtons.navSearch));
+    ImageView bookmarkButton = mSearchButtonFrame.findViewById(R.id.btn_bookmarks);
+    bookmarkButton.setImageDrawable(Graphics.tint(bookmarkButton.getContext(),
+                                                  R.drawable.ic_menu_bookmarks));
```

Comment:
```suggestion
    mSearchButtonFrame.findViewById(R.id.btn_bookmarks).setImageDrawable(Graphics.tint(bookmarkButton.getContext(), R.drawable.ic_menu_bookmarks));
```

Reply:
This refactoring is not possible as the `bookmarkButton` variable is used twice (`bookmarkButton.setImageDrawable` and `bookmarkButton.getContext()`).